app/views/self_service/advisers/_advisers_table.html.erb

```diff
@@ -22,6 +23,9 @@
               <%= link_to adviser.name, edit_self_service_firm_adviser_path(adviser.firm, adviser), class: 't-edit-link' %>
             </td>
             <td class="t-adviser-postcode" data-dough-filterable><%= adviser.postcode %></td>
+            <td>
+              <%= link_to 'Remove from Directory', self_service_firm_adviser_path(firm, adviser), method: :delete, data: {confirm: 'Are you sure?'}, :class => 'button button--small t-remove-button' %>
```

Comment:
jquery-ujs is being used for two things here.
* enabling the user to send a HTTP :delete to the server.
* the confirm part.

My confusion today was chasing the first point.

All of the html src produced by rails is fine, however jquery neglects to pick up (or hides) the HTTP_DELETE aspect of the link. This results in a Get request being sent to the server, which confuses the router as it tries to hit a `show` action rather than the `destroy` action.

Jquery-ujs enables (or pays attention to) the `delete` part of the request to make it to the server and can be processed correctly by the rails router.

With regards to the :confirm aspect this usually works with rails straight out of the box, however the jquery-ujs part wasn't enabled in this project yet.